blocks/models/huggingface/base.py

```diff
@@ -78,31 +79,33 @@ def load(self, pipeline_id: str, execution_order: int) -> int:
         self.pipeline_id = pipeline_id
         self.id += f"-{str(execution_order)}"
 
-        model = safe_load_pipeline(
-            f"{Const.output_pipelines_path}/{self.pipeline_id}/{self.id}"
-        )
-        if model:
-            self.model = model
+        local_path = f"{Const.output_pipelines_path}/{self.pipeline_id}/{self.id}"
+        remote_path = f"{self.config.user_name}/{self.id}"
+        pretrained_path = self.config.pretrained_model
+
+        if self.config.preferred_load_origin == Const.local:
```

Comment:
I'd propose to create an enum, that encompasses the options, as well as this function (that returns the order:)